services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceController.java

```diff
@@ -305,6 +308,19 @@ private Optional<DeferredVersionSwapService> createDeferredVersionSwapService()
     return Optional.empty();
   }
 
+  private Optional<ProtocolVersionDetectionService> createProtocolVersionDetectionService() {
+    if (multiClusterConfigs.isParent()) {
+      // Only create the ProtocolVersionDetectionService if this is a parent controller
+      // and the feature is enabled in the configuration.
+      Admin admin = controllerService.getVeniceHelixAdmin();
+      return Optional.of(
+          new ProtocolVersionDetectionService(
+              (VeniceParentHelixAdmin) admin,
+              multiClusterConfigs.getControllerConfig(multiClusterConfigs.getSystemSchemaClusterName())));
```

Comment:
Here is the problem.
multiClusterConfigs.getSystemSchemaClusterName() is only pointing to a dedicated cluster.

Reply:
How about changing it to:
```
              multiClusterConfigs.getControllerConfig(multiClusterConfigs.getControllerName())));
```